crates/pathfinder/src/storage/contract.rs

```diff
@@ -114,6 +114,55 @@ impl ContractCodeTable {
         Ok(Some(ContractCode { bytecode, abi }))
     }
 
+    pub fn get_class(
+        transaction: &Transaction,
+        hash: ClassHash,
+    ) -> anyhow::Result<Option<ContractClass>> {
+        let row = transaction
+            .query_row(
+                "SELECT bytecode, definition
+                FROM contract_code
+                WHERE hash = :hash",
+                named_params! {
+                    ":hash": &hash.0.to_be_bytes()
+                },
+                |row| {
+                    let bytecode: Vec<u8> = row.get("bytecode")?;
+                    let definition: Vec<u8> = row.get("definition")?;
+
+                    Ok((bytecode, definition))
+                },
+            )
+            .optional()?;
+
+        let (bytecode, definition) = match row {
+            None => return Ok(None),
+            Some((bytecode, definition)) => (bytecode, definition),
+        };
+
+        // It might be dangerious to not have some upper bound on the compressed size.
+        // someone could put a very tight bomb to our database, and then have it OOM during
+        // runtime, but if you can already modify our database at will, maybe there's more useful
+        // things to do.
```

Comment:
Are you referring to someone manually inserting into the database? In that case I think they could do whatever they want :D

Reply:
I wrote it, hasn't been discussed in the past.

Reply:
```suggestion
```

Not sure if it makes any sense to copy though.